packages/zoe/src/contracts/priceAggregator.js

```diff
@@ -343,79 +355,93 @@ const start = async (zcf, privateArgs) => {
 
-        // Support for notifiers that don't produce ratios, but need scaling for
-        // their numeric data.
-        const priceScale = parseRatio(scaleValueOut, brandOut);
-        const makeScaledRatioFromData = numericData => {
-          const ratio = makeRatioFromData(numericData);
-          return multiplyRatios(ratio, priceScale);
-        };
-
-        /**
-         * Adapt the notifier to push results.
-         *
-         * @param {UpdateRecord<OraclePriceSubmission>} param0
-         */
-        const recurse = ({ value, updateCount }) => {
-          if (!oracleNotifier || !updateCount) {
-            // Interrupt the cycle because we either are deleted or the notifier
-            // finished.
-            return;
-          }
-          // Queue the next update.
-          E(oracleNotifier).getUpdateSince(updateCount).then(recurse);
-
-          // See if we have associated parameters or just a raw value.
-          let result;
-          switch (typeof value) {
-            case 'number':
-            case 'bigint':
-            case 'string': {
-              result = makeScaledRatioFromData(value);
-              break;
+        if (oracleNotifier) {
+          // Support for notifiers that don't produce ratios, but need scaling for
+          // their numeric data.
+          const priceScale = parseRatio(scaleValueOut, brandOut);
+          const makeScaledRatioFromData = numericData => {
+            const ratio = makeRatioFromData(numericData);
+            return multiplyRatios(ratio, priceScale);
+          };
+
+          /**
+           * Adapt the notifier to push results.
+           *
+           * @param {UpdateRecord<OraclePriceSubmission>} param0
+           */
+          const recurse = ({ value, updateCount }) => {
+            if (!oracleNotifier || !updateCount) {
+              // Interrupt the cycle because we either are deleted or the notifier
+              // finished.
+              return;
             }
-            default: {
-              if (value && value.data !== undefined) {
-                result = {
-                  ...value,
-                  data: makeScaledRatioFromData(value.data),
-                };
-              } else {
-                result = value;
+            // Queue the next update.
+            E(oracleNotifier).getUpdateSince(updateCount).then(recurse);
+
+            // See if we have associated parameters or just a raw value.
+            let result;
+            switch (typeof value) {
+              case 'number':
+              case 'bigint':
+              case 'string': {
+                result = makeScaledRatioFromData(value);
+                break;
+              }
+              default: {
+                if (value && value.data !== undefined) {
+                  result = {
+                    ...value,
+                    data: makeScaledRatioFromData(value.data),
+                  };
+                } else {
+                  result = value;
```

Comment:
This much logic mixed in with I/O feels off. Consider hoisting this business of scaling and dealing with things other than ratios to module level functions and adding unit tests?

Reply:
Good call. I hoisted it up a bit. I could see it going out farther but I don't want to invest in this module too much because @michaelfig said it was made as a strawman and the chainlink one is the real one (so I restored it).